### PR TITLE
feat: pkg/util/gitlab Enhancement (sub task of pkg/util Enhancement #833) 

### PR DESCRIPTION
**Pull request by jxs1211**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
add ut for the package

## Related Issues
see https://github.com/devstream-io/devstream/issues/833 for the details

## New Behavior (screenshots if needed)
```sh
[going@dev gitlab]$ go tool cover -func=coverage.out
github.com/devstream-io/devstream/pkg/util/gitlab/ci_yml_templates.go:6:        GetGitLabCIGolangTemplate       75.0%
github.com/devstream-io/devstream/pkg/util/gitlab/commits.go:14:                CommitSingleFile                83.3%
github.com/devstream-io/devstream/pkg/util/gitlab/commits.go:35:                DeleteSingleFile                83.3%
github.com/devstream-io/devstream/pkg/util/gitlab/commits.go:55:                UpdateSingleFile                83.3%
github.com/devstream-io/devstream/pkg/util/gitlab/commits.go:76:                CommitMultipleFiles             90.0%
github.com/devstream-io/devstream/pkg/util/gitlab/gitlab.go:16:                 WithBaseURL                     100.0%
github.com/devstream-io/devstream/pkg/util/gitlab/gitlab.go:27:                 NewClient                       92.3%
github.com/devstream-io/devstream/pkg/util/gitlab/project.go:20:                CreateProject                   87.0%
github.com/devstream-io/devstream/pkg/util/gitlab/project.go:69:                DeleteProject                   75.0%
github.com/devstream-io/devstream/pkg/util/gitlab/project.go:78:                DescribeProject                 80.0%
github.com/devstream-io/devstream/pkg/util/gitlab/repository_files.go:12:       FileExists                      72.7%
total:                                                                          (statements)                    84.4%
```

### CODE REVIEW

**Comment by iyear** [comment]
Elegant unit tests! 🚀 I learned a lot about unit testing practices.

File: pkg/util/gitlab/repository_files.go [hunk 1]
```diff
@@ -13,7 +15,13 @@ func (c *Client) FileExists(project, branch, filename string) (bool, error) {
 	}
 
 	_, response, err := c.RepositoryFiles.GetFile(project, filename, getFileOptions)
-	if response.StatusCode == 404 {
+	if response.StatusCode == http.StatusBadRequest {
```
Comment (iyear) [inline pkg/util/gitlab/repository_files.go:18]: Perhaps a loop can be used to solve more situations in the future

Comment (iyear) [reply pkg/util/gitlab/repository_files.go:18]: @jxs1211 If you get stuck or don't have the time, I can provide you with a rough copy of the code.

```go
for _, code := range []int{http.StatusBadRequest, http.StatusUnauthorized, http.StatusNotFound} {
	if response.StatusCode == code {
		return false, nil
	}
}
```

Comment (jxs1211) [reply pkg/util/gitlab/repository_files.go:18]: @iyear obviously, this way is better than the previous one, let me update it.

**Review by iyear — APPROVED** [review]
Well Done! 🚀

**Review by hxcGit — APPROVED** [review]

**Pull request merged**